rama-net/src/tls/client/parser.rs

```diff
@@ -45,9 +45,9 @@ pub fn parse_client_hello(i: &[u8]) -> Result<ClientHello, OpaqueError> {
 ///
 /// Same as [`extract_sni_from_client_hello`] but handles the full handshake bytes, meaning
 /// the client hello record itself and the handshake header bytes in front of it.
-pub fn extract_sni_from_client_hello_handshake(i: &[u8]) -> Result<Host, OpaqueError> {
+pub fn extract_sni_from_client_hello_handshake(i: &[u8]) -> Result<Domain, OpaqueError> {
```

Comment:
feel free to already in this PR make this `Option<Domain>` and the other code you were planning to do normally in a separate PR, might be easier for you? I'm fine for you to do it later in a follow up PR as well, up to you :)